common/src/app/common/geom/shapes/flex_layout/lines.cljc

```diff
@@ -0,0 +1,312 @@
+
+                next-min-width   (+ child-margin-width (if fill-width? child-min-width child-width))
+                next-min-height  (+ child-margin-height (if fill-height? child-min-height child-height))
+                next-max-width   (+ child-margin-width (if fill-width? child-max-width child-width))
+                next-max-height  (+ child-margin-height (if fill-height? child-max-height child-height))
+
+                next-line-min-width  (+ line-min-width  next-min-width  (* layout-gap-row num-children))
+                next-line-min-height (+ line-min-height next-min-height (* layout-gap-col num-children))]
+            (if (and (some? line-data)
+                     (or (not wrap?)
+                         (and row? (<= next-line-min-width layout-width))
+                         (and col? (<= next-line-min-height layout-height))))
+
+              [{:line-min-width  (if row? (+ line-min-width next-min-width) (max line-min-width next-min-width))
+                :line-max-width  (if row? (+ line-max-width next-max-width) (max line-max-width next-max-width))
+                :line-min-height (if col? (+ line-min-height next-min-height) (max line-min-height next-min-height))
+                :line-max-height (if col? (+ line-max-height next-max-height) (max line-max-height next-max-height))
+                :num-children    (inc num-children)
+                :children-data   (conjv children-data child-data)}
+               result]
+
+              [{:line-min-width  next-min-width
+                :line-min-height next-min-height
+                :line-max-width  next-max-width
+                :line-max-height next-max-height
+                :num-children    1
+                :children-data   [child-data]}
+               (cond-> result (some? line-data) (conj line-data))])))
+
+        [line-data layout-lines] (reduce calculate-line-data [nil []] children)]
+
+    (cond-> layout-lines (some? line-data) (conj line-data))))
+
+
+(defn add-space-to-items
+  ;; Distributes the remainder space between the lines
+  [prop prop-min prop-max to-share items]
+  (let [num-items (->> items (remove #(mth/close? (get % prop) (get % prop-max))) count)
+        per-line-target (/ to-share num-items)]
+    (loop [current (first items)
+           items   (rest items)
+           remainder to-share
+           result []]
+      (if (nil? current)
+        [result remainder]
+        (let [cur-val (or (get current prop) (get current prop-min) 0)
+              max-val (get current prop-max)
+              cur-inc (if (> (+ cur-val per-line-target) max-val)
+                        (- max-val cur-val)
+                        per-line-target)
+              current (assoc current prop (+ cur-val cur-inc))
+              remainder (- remainder cur-inc)
+              result (conj result current)]
+          (recur (first items) (rest items) remainder result))))))
```

Comment:
This is just FYI, that there are a way to write loop without maintaining the both: first and rest on the bindings. Just maintaining the seq. Example:

```clojure
(loop [items     (seq items)
       remainder to-share
       result    []]
  (if-let [current (first items)]
    (let [cur-val (or (get current prop) (get current prop-min) 0)
          max-val (get current prop-max)
          cur-inc (if (> (+ cur-val per-line-target) max-val)
                    (- max-val cur-val)
                    per-line-target)
          current (assoc current prop (+ cur-val cur-inc))]
      (recur (rest items)
             (- remainder cur-inc)
             (conj result current)))
    [result remainder]))
```

I don't suggest to change, this is just a informative comment.